# AdvCiv-SAS reachable-history commit diff
# Email privacy: email-shaped addresses redacted from generated commit-history text.
# History metadata layout: 3 (history-segment label; title-only diff header; full message redirected to anonymized Git logs).
# Cache format: 2
# Cache policy: 8c87d6e4bd0f
# Practical commit count: 4639
# Commit: 10c14b0736d1ae68ab1fd7fa03c8d1823aa486bf
# History segment: AdvCiv-SAS branch history
# Parent used for diff: 2dd4efa3da2d5ef854a56c2be38aaa73fae85b21
# Title: add Bombard column for all units, further beautify, make code (quite a bit) more reliable or/and other changes/tweaks/fixes
# Full commit message/metadata: see _1_AdvCiv-SAS/git_logs/git_log_anonymized_email_003_AdvCiv-SAS.txt.
# History segments describe ancestry/log partition, not authorship: SASBranch includes SAS work plus later upstream AdvCiv commits merged/imported into this branch.
# Practical commit counts can repeat on divergent/merged history; the full Git SHA is the canonical unique commit identifier.
# Diff policy: preserve textual source/docs/config history; summarize known generated/binary/noisy or exceptionally huge file patches.
# Git textconv/external diff drivers are disabled; redundant document formats such as ODT/PDF are summarized rather than converted.

[CHANGE SUMMARY - NUMSTAT + RENAME/MODE SUMMARY]
71	48	Assets/Python/Contrib/Sevopedia/SevoPediaUnitChart.py

[PATCHES]

diff --git a/Assets/Python/Contrib/Sevopedia/SevoPediaUnitChart.py b/Assets/Python/Contrib/Sevopedia/SevoPediaUnitChart.py
index b733e15b1..58ec98ca3 100644
--- a/Assets/Python/Contrib/Sevopedia/SevoPediaUnitChart.py
+++ b/Assets/Python/Contrib/Sevopedia/SevoPediaUnitChart.py
@@ -16,7 +16,7 @@ class SevoPediaUnitChart:
 
 		self.X_TABLE = self.top.X_PEDIA_PAGE
 		self.Y_TABLE = self.top.Y_PEDIA_PAGE
-		self.W_TABLE = 900
+		self.W_TABLE = 1000
 		self.H_TABLE = self.top.H_PEDIA_PAGE
 
 		self.MARGIN = 20
@@ -33,7 +33,7 @@ class SevoPediaUnitChart:
 		screen = self.top.getScreen()
 		table = self.top.getNextWidgetName()
 
-		nColumns = 7
+		nColumns = 8
 		# <!-- custom: blue is more readable than standard i find, imported
 		# from base AdvCiv and modified with a similar kind of purpose
 		# -->
@@ -46,8 +46,10 @@ class SevoPediaUnitChart:
 		screen.addTableControlGFC(table, nColumns, self.X_TABLE + self.MARGIN, self.Y_TABLE + self.MARGIN + 5, self.W_TABLE - (self.MARGIN * 2), self.H_TABLE - (self.MARGIN * 2) - 5, True, False, 32,32, TableStyles.TABLE_STYLE_EMPTY)
 		screen.enableSort(table)
 
-		if self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR'):
-			szMove = "Range"
+		selfIsAirUnit = ( (self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR_BOMBER')) or (self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR_FIGHTER')) )
+
+		if selfIsAirUnit:
+			szMove = u"Range"
 		else:
 			# <!-- custom: imported from based AdvCiv and modified in a
 			# similar way as for/in other parts of this code, to display
@@ -55,34 +57,44 @@ class SevoPediaUnitChart:
 			#szMove = "Moves"
 			szMove = u"%c" % CyGame().getSymbolID(FontSymbols.MOVES_CHAR)
 
-		if self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR') or self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_NAVAL') or self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_SIEGE'):
+		# <!-- custom: display more potentially useful information,
+		# useful in AdvCiv-SAS and/or maybe other mods as well,
+		# also so that i don't have to maintain it everytime i make
+		# a micro (or big) change, more versatile and flexible this
+		# way too.
+		# -->
+		# if self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR') or self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_NAVAL') or self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_SIEGE'):
 			# <!-- custom: imported and modified from base advciv in a similar way -->
-			#szSpecial = "Bombard"
-			szSpecial = u"%c" % CyGame().getSymbolID(FontSymbols.DEFENSE_CHAR)
-		else:
-			szSpecial = "1st Strike"
+			#szFirstStrike = "Bombard"
+		szBombard = u"%c" % CyGame().getSymbolID(FontSymbols.DEFENSE_CHAR)
 
-		if self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR'):
-			szWithdraw = "Evade"
+		szFirstStrike = u"1st Strike"
+
+		if selfIsAirUnit:
+			szWithdraw = u"Evade"
 
 		else:
-			szWithdraw = "Withdraw"
+			szWithdraw = u"Withdraw"
+
+		iWidthNames = 270
+		iWidthNums = ((self.W_TABLE - (self.MARGIN * 2) - iWidthNames) / (nColumns - 1))
 
-		iWidth = ((self.W_TABLE - 180) / (nColumns - 1))
-		screen.setTableColumnHeader(table, 0, u"<font=2>" + gc.getUnitCombatInfo(self.iGroup).getDescription() + "</font>", 182)
+
+		screen.setTableColumnHeader(table, 0, u"<font=2>" + gc.getUnitCombatInfo(self.iGroup).getDescription() + u"</font>", iWidthNames)
 		# <!-- custom: add strength icon, imported from base AdvCiv code:
 		# https://github.com/f1rpo/AdvCiv/blob/master/Assets/Python/Contrib/Sevopedia/SevoPediaUnitChart.py
 		# and modified for AdvCiv-SAS to display the icon instead of text -->
 		# -->
-		#screen.setTableColumnHeader(table, 1, u"<font=2>" + "Strength" + "</font>", iWidth)
-		screen.setTableColumnHeader(table, 1, u"%c" % CyGame().getSymbolID(FontSymbols.STRENGTH_CHAR), iWidth)
-		screen.setTableColumnHeader(table, 2, u"<font=2>" + szMove + "</font>", iWidth)
-		screen.setTableColumnHeader(table, 3, u"<font=2>" + szSpecial + "</font>", iWidth)
-		screen.setTableColumnHeader(table, 4, u"<font=2>" + "Collateral" + "</font>", iWidth)
-		screen.setTableColumnHeader(table, 5, u"<font=2>" + szWithdraw + "</font>", iWidth)
+		#screen.setTableColumnHeader(table, 1, u"<font=2>" + "Strength" + "</font>", iWidthNums)
+		screen.setTableColumnHeader(table, 1, u"%c" % CyGame().getSymbolID(FontSymbols.STRENGTH_CHAR), iWidthNums)
+		screen.setTableColumnHeader(table, 2, u"<font=2>" + szMove + u"</font>", iWidthNums)
+		screen.setTableColumnHeader(table, 3, u"<font=2>" + szFirstStrike + u"</font>", iWidthNums)
+		screen.setTableColumnHeader(table, 4, u"<font=2>" + szBombard + u"</font>", iWidthNums)
+		screen.setTableColumnHeader(table, 5, u"<font=2>" + u"Collateral" + u"</font>", iWidthNums)
+		screen.setTableColumnHeader(table, 6, u"<font=2>" + szWithdraw + u"</font>", iWidthNums)
 		# <!-- custom: imported and modified from base advciv in a similar way -->
-		#screen.setTableColumnHeader(table, 6, u"<font=2>" + "Cost" + "</font>", iWidth)
-		screen.setTableColumnHeader(table, 6, u"%c" % gc.getYieldInfo(YieldTypes.YIELD_PRODUCTION).getChar(), iWidth)
+		#screen.setTableColumnHeader(table, 6, u"<font=2>" + "Cost" + "</font>", iWidthNums)
+		screen.setTableColumnHeader(table, 7, u"%c" % gc.getYieldInfo(YieldTypes.YIELD_PRODUCTION).getChar(), iWidthNums)
 
 		for iUnit in xrange(gc.getNumUnitInfos()):
 			UnitInfo = gc.getUnitInfo(iUnit)
@@ -96,64 +108,75 @@ class SevoPediaUnitChart:
 				# Combat Strength
 				iCol += 1
 				if UnitInfo.getAirCombat() > 0:
-					szCombat = u"%d" % UnitInfo.getAirCombat()
+					szCombatNum = u"%d" % UnitInfo.getAirCombat()
 				else:
-					szCombat = u"%d" % UnitInfo.getCombat()
+					szCombatNum = u"%d" % UnitInfo.getCombat()
 
-				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szCombat + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
+				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szCombatNum + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
 
 				# Movement
 				iCol += 1
 				if UnitInfo.getAirRange() > 0:
-					szMoves = u"%d" % UnitInfo.getAirRange()
+					szMovesNum = u"%d" % UnitInfo.getAirRange()
+				else:
+					szMovesNum = u"%d" % UnitInfo.getMoves()
+
+				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szMovesNum + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
+
+				# First Strikes
+				iCol += 1
+				if UnitInfo.getFirstStrikes() > 0:
+					szBombardRate = u"%d" % UnitInfo.getFirstStrikes()
 				else:
-					szMoves = u"%d" % UnitInfo.getMoves()
+					szBombardRate = u""
 
-				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szMoves + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
+				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szBombardRate + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
 
-				# Bombard/First Strikes
+				# Bombard
 				iCol += 1
 				if UnitInfo.getBombardRate() > 0:
-					szBombard = u"%d%%" % UnitInfo.getBombardRate()
+					szBombardRate = u"%d%%" % UnitInfo.getBombardRate()
 				elif UnitInfo.getBombRate() > 0:
-					szBombard = u"%d%%" % UnitInfo.getBombRate()
-				elif UnitInfo.getFirstStrikes() > 0 and not (self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR') or self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_NAVAL') or self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_SIEGE')):
-					szBombard = u"%d" % UnitInfo.getFirstStrikes()
+					szBombardRate = u"%d%%" % UnitInfo.getBombRate()
+				# <!-- custom: keeping this beautification as it is a lot more
+				# useful/readable than having tons of 0s, anyways -->
 				else:
-					szBombard = u""
+					szBombardRate = u""
 
-				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szBombard + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
+				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szBombardRate + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
 
 				# Collateral
 				iCol += 1
 				if UnitInfo.getCollateralDamage() > 0:
-					szCollateral = u"%d%% (%d)" % (UnitInfo.getCollateralDamageLimit(), UnitInfo.getCollateralDamageMaxUnits())
+					szCollateralRate = u"%d%% (%d)" % (UnitInfo.getCollateralDamageLimit(), UnitInfo.getCollateralDamageMaxUnits())
 				else:
-					szCollateral = u""
+					szCollateralRate = u""
 
-				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szCollateral + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
+				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szCollateralRate + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
 
 				# Withdrawal
 				iCol += 1
-				if self.iGroup == gc.getInfoTypeForString('UNITCOMBAT_AIR') and UnitInfo.getEvasionProbability() > 0:
-					szWithdrawal = u"%d%%" % UnitInfo.getEvasionProbability()
+				if selfIsAirUnit and UnitInfo.getEvasionProbability() > 0:
+					szWithdrawalRate = u"%d%%" % UnitInfo.getEvasionProbability()
 				elif UnitInfo.getWithdrawalProbability() > 0:
-					szWithdrawal = u"%d%%" % UnitInfo.getWithdrawalProbability()
+					szWithdrawalRate = u"%d%%" % UnitInfo.getWithdrawalProbability()
 				else:
-					szWithdrawal = u""
+					szWithdrawalRate = u""
 
-				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szWithdrawal + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
+				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szWithdrawalRate + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
 
 				# Cost
 				iCol += 1
 				if UnitInfo.getProductionCost() < 0:
-					szCost = CyTranslator().getText("TXT_KEY_NON_APPLICABLE", ())
+					szCostNum = CyTranslator().getText("TXT_KEY_NON_APPLICABLE", ())
 				else:
-					szCost = u"%d" % UnitInfo.getProductionCost()
+					szCostNum = u"%d" % UnitInfo.getProductionCost()
 
-				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szCost + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
+				screen.setTableInt(table, iCol, iRow, u"<font=3>" + szCostNum + u"</font>", "", WidgetTypes.WIDGET_GENERAL, -1, -1, CvUtil.FONT_CENTER_JUSTIFY)
 
 
 
-	def handleInput(self, inputClass):
-		return 0
+	# <!-- custom: seems we never call this function, commenting it out,
+	# may even be deleted maybe(?) -->
+	#def handleInput(self, inputClass):
+	#	return 0

[PATCH COVERAGE]
patchFiles=1 embeddedFiles=1 omittedFiles=0 embeddedPatchBytes=10403
